pulsar-client-api/src/main/java/org/apache/pulsar/client/api/transaction/TxnID.java

```diff
@@ -39,14 +43,14 @@ public class TxnID implements Serializable {
      *
      * @serial
      */
-    private final long mostSigBits;
+    private long mostSigBits;
```

Comment:
Why remove the final modifier?

Same for `leastSigBits`

Reply:
We need NoArgsConstructor to make it can be deserialized as a schema.

Reply:
OK

Reply:
I think the issue is we are using by `TransactionBufferSnapshotIndexes` right? It's better to add a new one in the broker side like `TxnIdData`.

Here we will change a public API for users. It's better to keep it simple and understandable. I don't think the user will need the new constructor of TxnID.

And allow user to update the `mostSigBits` and `leastSigBits` is also not good. We should avoid it.